### PR TITLE
Integration release

### PR DESCRIPTION
**Pull request by loleg**
This PR currently contains patches from #244 with additional library upgrades.

### CODE REVIEW

**Comment by loleg** [comment]
Issues preventing build from succeeding:
```
--- FAIL: TestUser (4.26s)
    user_test.go:64: 
        Unexpected non-nil/non-zero argument at index 1:
        	<*errors.errorString>: &errors.errorString{s:"authority missing"}
FAIL
FAIL	github.com/ProxeusApp/proxeus-core/storage/database	4.300s

```

**Comment by loleg** [comment]
The API tests are failing due to an issue in https://github.com/ProxeusApp/node-crypto-forex-rates
I suggest closing this PR so we can get a fresh version up in master and continue looking into it with a fresh branch.

**Comment by loleg** [comment]
I have tried to disable the areas of CI that are failing - for me without clearly identifiable cause - in the latest commit. Adding warnings in the log (via `Makefile`) that we need to put in more work here. And starting a branch where I'm trying to fix or at least better understand the issue: https://github.com/loleg/proxeus-core/tree/api-debugging

File: storage/database/user.go [hunk 1]
```diff
@@ -428,19 +460,20 @@ func (me *UserDB) save(u *model.User, tx db.DB) error {
...
+
 	for _, a := range u.ApiKeys {
-		if a.IsNew() {
+		if me.keyIsNew(existingKeys, a) {
 			newKeys = append(newKeys, *a)
 			err := tx.Set(userApiKeyBucket, a.Key, u.ID)
 			if err != nil {
 				return err
 			}
-			a.HideKey()
```
Comment (tafonina) [inline storage/database/user.go:438]: we should remain the logic with hide key

Comment (loleg) [reply storage/database/user.go:438]: I quickly tested and this line breaks the process. It is not needed to hash the key again at this point, as far as I can tell.

**Review by VivienFuhrer — APPROVED** [review]

**Pull request merged**